homeassistant/components/supla/__init__.py

```diff
@@ -0,0 +1,157 @@
+"""Support for the Supla devices."""
+import logging
+from typing import Optional
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_ACCESS_TOKEN
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pysupla==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'supla'
+
+CONF_SERVER = 'server'
+CONF_SERVERS = 'servers'
+
+SUPLA_FUNCTION_HA_CMP_MAP = {
+    'CONTROLLINGTHEROLLERSHUTTER': 'cover'
+}
+SUPLA_CHANNELS = 'supla_channels'
+SUPLA_SERVERS = 'supla_servers'
+
+SERVER_CONFIG = vol.Schema({
+    vol.Required(CONF_SERVER): cv.string,
+    vol.Required(CONF_ACCESS_TOKEN): cv.string
+}, extra=vol.ALLOW_EXTRA)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_SERVERS):
+            vol.All(cv.ensure_list, [SERVER_CONFIG])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def discover_devices(hass, hass_config):
```

Comment:
Please move this below `setup` so that the `CONFIG_SCHEMA` and `setup` are close. It's easier to read like that and it's our standard.

Reply:
Fixed.